### ISSUE TITLE
active support cannot load bigdecimal - gets an undefined symbolm error

### ISSUE THREAD

**Issue by Hellfire01** [rails/rails#34822]
### Steps to reproduce
Installed Ruby on an archlinux / manjaro environment ( up to date ) in a virtual machine
Installed the rails gem with a `gem install rails`
Tried a `require 'rails'` in both irb and a file

### Expected behavior
No issues loading the gem

### Actual behavior
The gem cannot be loaded. There seems to be an issue when trying to load the bigdecimal gem.
I have no issue when trying to load the bigdecimal gem directly.
The error is the following :
```
irb(main):001:0> require 'rails'
Traceback (most recent call last):
       16: from /home/matthieu/.gem/ruby/2.5.0/gems/railties-5.2.2/lib/rails/configuration.rb:4:in `<top (required)>'
       15: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
       14: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
       13: from /home/matthieu/.gem/ruby/2.5.0/gems/activesupport-5.2.2/lib/active_support/core_ext/object.rb:13:in `<top (required)>'
       12: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
       11: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
       10: from /home/matthieu/.gem/ruby/2.5.0/gems/activesupport-5.2.2/lib/active_support/core_ext/object/json.rb:8:in `<top (required)>'
        9: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
        8: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
        7: from /home/matthieu/.gem/ruby/2.5.0/gems/activesupport-5.2.2/lib/active_support/core_ext/big_decimal/conversions.rb:4:in `<top (required)>'
        6: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
        5: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
        4: from /home/matthieu/.gem/ruby/2.5.0/gems/bigdecimal-1.4.2/lib/bigdecimal/util.rb:9:in `<top (required)>'
        3: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:39:in `require'
        2: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:135:in `rescue in require'
        1: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:135:in `require'
LoadError (/home/matthieu/.gem/ruby/2.5.0/gems/bigdecimal-1.4.2/lib/bigdecimal/util.so: undefined symbol: rmpd_util_str_to_d - /home/matthieu/.gem/ruby/2.5.0/gems/bigdecimal-1.4.2/lib/bigdecimal/util.so)
```

### System configuration
**Rails version**: rails (5.2.2)
**Ruby version**: Ruby 2.5.3p105

I am at a loss on what to do to solve this issue could you please help me ?

**Comment by y-yagi** [comment]
It seems bigdecimal's issue. 
Please report to [bigdecimal's repo](https://github.com/ruby/bigdecimal/issues/new). Thanks.

**Comment by gleopoldo** [comment]
Hi there!
I've fall into this issue as well. I catch this when trying to upgrade the ruby version in an old internal gem from ruby 2.3 to a newer one.

While running my suite with this configuration in the Gemfile, I had the same error:

```
ruby '~> 2.4'

gem 'activerecord', '~> 4.2.7'
gem 'activemodel', '~> 4.2.7'
gem 'activesupport', '~> 4.2.7'

gemspec
```

This was the error I was facing:

```
NoMethodError: undefined method `new' for BigDecimal:Class
/usr/local/bundle/gems/activesupport-4.2.11.1/lib/active_support/core_ext/object/duplicable.rb:111:in `<class:BigDecimal>'
/usr/local/bundle/gems/activesupport-4.2.11.1/lib/active_support/core_ext/object/duplicable.rb:106:in `<top (required)>'
/usr/local/bundle/gems/activesupport-4.2.11.1/lib/active_support/dependencies.rb:274:in `require'
/usr/local/bundle/gems/activesupport-4.2.11.1/lib/active_support/dependencies.rb:274:in `block in require'
/usr/local/bundle/gems/activesupport-4.2.11.1/lib/active_support/dependencies.rb:240:in `load_dependency'
/usr/local/bundle/gems/activesupport-4.2.11.1/lib/active_support/dependencies.rb:274:in `require'
/usr/local/bundle/gems/activerecord-4.2.11.1/lib/active_record/base.rb:16:in `<top (required)>'
/usr/local/bundle/gems/activerecord-import-0.10.0/lib/activerecord-import.rb:1:in `require'
/usr/local/bundle/gems/activerecord-import-0.10.0/lib/activerecord-import.rb:1:in `<top (required)>'
/builds/project-0/lib/saas_smtp_models/models.rb:2:in `require'
/builds/project-0/lib/saas_smtp_models/models.rb:2:in `<top (required)>'
/builds/project-0/lib/saas_smtp_models.rb:16:in `require_relative'
/builds/project-0/lib/saas_smtp_models.rb:16:in `<top (required)>'
/builds/project-0/lib/tasks/saas_smtp_models.rake:1:in `require'
/builds/project-0/lib/tasks/saas_smtp_models.rake:1:in `<top (required)>'
/builds/project-0/lib/saas_smtp_models/tasks.rb:1:in `load'
/builds/project-0/lib/saas_smtp_models/tasks.rb:1:in `<top (required)>'
/builds/project-0/Rakefile:2:in `require'
/builds/project-0/Rakefile:2:in `<top (required)>'
/usr/local/bundle/gems/rake-13.0.1/exe/rake:27:in `<top (required)>'
/usr/local/bin/bundle:23:in `load'
/usr/local/bin/bundle:23:in `<main>'
(See full trace by running task with --trace)
```

I finally found a solution when reading this issue at bigdecimal (https://github.com/ruby/bigdecimal/issues/127). Seems that they had released a new major recently (they released 2.0.0 10 days ago) and it's being installed along with activesupport somehow.

The [bigdecimal readme](https://github.com/ruby/bigdecimal#which-version-should-you-select) has been updated with the deprecation information.

I didn't find it in your gemspec. However, adding this line to my Gemfile, as suggested in the linked issue, has resolved the issue:

```
gem 'bigdecimal', '~> 1.4'
```

Do you guys think that we can either enforce the usage of 1.X series of bigdecimal, or apply the needed fixes in order to make it compliant with the 2.X version of bigdecimal?

**Comment by chai2** [comment]
`gem 'bigdecimal', '1.4.2'` fixed the issue for me on Ruby 2.7.0

**Comment by Firooz-Bright** [comment]
warning: BigDecimal.new is deprecated; use BigDecimal() method instead.
 I used the gem 'bigdecimal', '1.4.2'. worked with the above warning
